src/hooks/routing/useRouterTrade.ts

```diff
@@ -72,31 +71,21 @@ export function useRouterTrade(
   }, [fulfilledTimeStamp, pollingInterval, queryArgs, trigger])
   useTimeout(request, 200)
 
-  const quote = typeof data === 'object' ? data : undefined
-  const trade = useMemo(() => {
-    const routes = computeRoutes(currencyIn, currencyOut, tradeType, quote)
-    if (!routes || routes.length === 0) return
-    try {
-      return transformRoutesToTrade(routes, tradeType)
-    } catch (e: unknown) {
-      console.debug('transformRoutesToTrade failed: ', e)
-      return
-    }
-  }, [currencyIn, currencyOut, quote, tradeType])
-  const isValidBlock = useIsValidBlock(Number(quote?.blockNumber))
+  const tradeResult: TradeResult | undefined = typeof data === 'object' ? data : undefined
+  const isValidBlock = useIsValidBlock(Number(tradeResult?.blockNumber))
   const isValid = currentData === data && isValidBlock
-  const gasUseEstimateUSD = useStablecoinAmountFromFiatValue(quote?.gasUseEstimateUSD)
+  const gasUseEstimateUSD = useStablecoinAmountFromFiatValue(tradeResult?.gasUseEstimateUSD)
 
   return useMemo(() => {
     if (!amountSpecified || isError || queryArgs === skipToken) {
       return TRADE_INVALID
     } else if (data === NO_ROUTE) {
       return TRADE_NOT_FOUND
-    } else if (!trade) {
+    } else if (!tradeResult?.trade) {
       return TRADE_LOADING
     } else {
       const state = isValid ? TradeState.VALID : TradeState.LOADING
-      return { state, trade, gasUseEstimateUSD }
+      return { state, trade: tradeResult?.trade, gasUseEstimateUSD }
```

Comment:
this reads like `TradeState.LOADING` can be the result of two logic paths, is that right? Having the last `else` case include a ternary when we're concluding an existing if/else block feels messy.

Reply:
can take a look at refactoring the loading states in another pr!